src/main/java/io/tiledb/spark/SubArrayRanges.java

```diff
@@ -255,6 +255,37 @@ public boolean splittable() {
     return newSubarrays;
   }
 
+    /**
+     * Splits the subarray ranges into partitions
+     * @param partitions The number of partitions
+     * @return A list of SubArrayRanges
+     * @throws TileDBError
+     */
+  public List<SubArrayRanges> splitToPartitions(int partitions) throws TileDBError {
+    List<SubArrayRanges> newSubarrays = new ArrayList<>();
+
+    long actualWidth = ranges.get(0).width().longValue();
+    double exactEffectiveWidth = actualWidth * 1.0 / partitions;
+
+    long partitionWidth = (long) exactEffectiveWidth;
+
+    if (exactEffectiveWidth < 1) {
+      partitions = (int) actualWidth;
+      partitionWidth = 1;
+    }
+
+    // Split the first dimension
+    if (ranges.get(0).splittable()) {
+      for (Range newRange : ranges.get(0).splitRangeToPartitions(partitions, partitionWidth)) {
+        List<Range> dimRanges = new ArrayList<>();
+        dimRanges.add(newRange);
```

Comment:
Since you are only splitting the first dimension, don't you need to add the other dimension's ranges?
```
// Start at 1 to skip dim0
for (int i =1; i < ranges.length(); i++) {
dimRanges.add(ranges[i])
}
```

Reply:
You mean we should completely remove that block and just set keep ranges for the first dimension only?

Reply:
The SubArrayRanges class has the list of ranges for all dimensions. In your code here you are only collecting back the first dimensions ranges you've split but if `ranges.size() > 1` you are missing all the remaining dimension's ranges.

I.e. if we have a 2 dimensional array, the `SubArrayRanges.ranges` might equal something like `[[1,10], [1000, 1000]] `. If you split that into 2 partitions your current code yields:
`[[1, 5]]` and `[[6,10]]`. You are not adding back the `[1000, 1000]` range for dimension 2.

Reply:
Fixed that.